discovery/utils/visualizer.py

```diff
@@ -45,45 +44,67 @@ class Visualizer:
+    def _generate_root_graph(self):
         start_graph = graphviz.Graph(name="cluster_root", comment='File Visualization',
                                      node_attr={'shape': 'plaintext'})
         start_graph.attr(label="")
         start_graph.engine = self.engine
         return start_graph
 
-    def _draw_metadata(self, metadata: Metadata):
-        cols = self._draw_columns(metadata)
-        filled_table = self._generate_table(metadata.filepath, cols)
-        self.working_node.graph.node(metadata.filepath, filled_table)
+    def draw(self, metadata: [Metadata], filename: str):
+        self._draw_metadata(metadata)
+        # self.draw_relationships(metadata)
+        self._finalize_result_graph(filename)
 
-    def _draw_columns(self, metadatum):
-        col_row: str = ''
+    def _draw_metadata(self, metadata):
+        for datum in metadata:
+            self.working_node = self._determine_working_node(datum.file_path)
+            self._draw_metadatum(datum)
+
+    # Don't use it, not implemented properly
+    def _draw_relationships(self, metadata):
+        for metadatum in metadata:
+            for column in metadatum.columns:
+                for relationship in column.relationships:
+                    self.root.graph.edge(f"{metadatum.hash}:{column.name}",
+                                         f"{relationship.target_file_hash}:{relationship.target_column_name}",
+                                         str(relationship.certainty)
+                                         )
+
+    def _finalize_result_graph(self, output_filename):
+        self.root.recursively_append_subgraphs()
+        self.root.graph.view(output_filename)
+
+    def _draw_metadatum(self, metadatum: Metadata):
+        columns = self._draw_table_columns(metadatum)
+        filled_table = self._draw_filled_metadatum_table(metadatum.file_path, columns)
+        self.working_node.graph.node(str(metadatum.hash), filled_table)
+
+    # TODO: make it more generic
+    def _draw_table_columns(self, metadatum):
+        col_rows: str = ""
         for column in metadatum.columns:
-            col_row += "<TR><TD>{}</TD> <TD>{}</TD> <TD>{}</TD> <TD>{}</TD> <TD>{}</TD></TR>" \
+            col_rows += '<TR><TD>{}</TD> <TD>{}</TD> <TD>{}</TD> <TD>{}</TD> <TD>{}</TD></TR>' \
                 .format(column.name, column.col_type,
                         (column.mean if column.mean is not None else "NA"),
-                        column.min, column.max)
-        return col_row
+                        column.minimum, column.maximum)
+        return col_rows
 
-    def draw(self, metadata: [Metadata], filename: str):
-        for datum in metadata:
-            self.working_node = self._determine_working_node(datum.filepath)
-            self._draw_metadata(datum)
-        self.root.recursively_add_subgraphs()
-        self.root.graph.view(filename)
+    # TODO: break down mean string generator
+    def _draw_column_row(self, column):
```

Comment:
Unused?
it would probably make more sense to have the string as a global var or class attribute that gets formatted

Reply:
Graphviz will probably sooner or later be removed anyway